.github/workflows/update_software_list.yml

```
@@ -0,0 +1,111 @@
+    # Don't run if we're seeing an update push
+    if: github.actor != needs.setup.outputs.git_user
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.CISAGOVBOT_PAT }}
+      - id: setup-python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
+      - uses: actions/cache@v2
+        env:
+          BASE_CACHE_KEY: "${{ github.job }}-${{ runner.os }}-\
+            py${{ steps.setup-python.outputs.python-version }}-"
+        with:
+          path: |
+            ${{ env.PIP_CACHE_DIR }}
+          key: "${{ env.BASE_CACHE_KEY }}\
+            ${{ hashFiles('.github/workflows/update_software_list.yml') }}-\
+            ${{ hashFiles('config/requirements.txt') }}"
+          restore-keys: |
+            ${{ env.BASE_CACHE_KEY }}
+      - name: Update Python base packages
+        run: python -m pip install --upgrade pip setuptools wheel
+      - name: Install dependencies
+        run: pip install --upgrade --requirement config/requirements.txt
+      - name: Create the branch for test validation
+        run: git switch --create ${{ needs.setup.outputs.testing_branch }}
+      - name: Generate a normalized YAML file
+        run: normalize-yml data/cisagov.yml > normalized.yml
+      - name: Generate a Markdown table from the normalized YAML file
+        run: yml2md normalized.yml > table_data.md
+      - name: Generate a new software list from the updated data
+        run: md-from-template config/SOFTWARE-LIST.tpl.md table_data.md > SOFTWARE-LIST.md
+      - id: commit-for-testing
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          branch: ${{ needs.setup.outputs.testing_branch }}
+          commit_message: Update the software list
+          commit_user_name: ${{ needs.setup.outputs.git_user }}
+          commit_user_email: ${{ needs.setup.outputs.git_email }}
+          commit_author: ${{ needs.setup.outputs.git_author }}
+          file_pattern: SOFTWARE-LIST.md
+  merge_list_update:
+    runs-on: ubuntu-latest
+    needs:
+      - setup
+      - generate_list_update
+    if: needs.generate_list_update.outputs.updated_list == 'true'
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.CISAGOVBOT_PAT }}
+      - name: Configure git
+        run: |
+          git config user.name "${{ needs.setup.outputs.git_user }}"
+          git config user.email "${{ needs.setup.outputs.git_email }}"
+      - uses: lewagon/wait-on-check-action@v1.0.0
+        with:
+          check-name: lint
+          ref: ${{ needs.setup.outputs.testing_branch }}
+          repo-token: ${{ github.token }}
+      - name: Merge the testing branch
+        run: |
+          git fetch
+          git merge origin/${{ needs.setup.outputs.testing_branch }}
+          git push
+      - name: Cleanup testing branch
+        run: git push -d origin ${{ needs.setup.outputs.testing_branch }}
```

Prefer full name of switch:
```suggestion
        run: git push --delete origin ${{ needs.setup.outputs.testing_branch }}
```

---

Good catch!